spec/requests/api/v1/invoices/invoice_request_spec.rb

```diff
@@ -1,11 +1,150 @@
+    get "/api/v1/invoices/find?created_at=15 May 2017"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(id)
+  end
+  it "finds one invoice by updated_at" do
+    id = create(:invoice, updated_at: "15 May 2017").id
+    create(:invoice, updated_at: "15 May 2017")
+    get "/api/v1/invoices/find?updated_at=15 May 2017"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(id)
+  end
+  it "finds multiple invoices by customer_id" do
+    create_list(:customer, 2)
+    create_list(:invoice, 3, customer_id: Customer.first.id)
+    create(:invoice, customer_id: Customer.second.id)
+    get "/api/v1/invoices/find_all?customer_id=#{Customer.first.id}"
+
+    expect(response).to be_success
+
+    invoices = JSON.parse(response.body)
+    expect(invoices.count).to eq(3)
+    expect(invoices.sample["customer_id"]).to eq(Customer.first.id)
+  end
+  it "finds multiple invoices by merchant_id" do
+    create_list(:merchant, 2)
+    create_list(:invoice, 3, merchant_id: Merchant.first.id)
+    create(:invoice, merchant_id: Merchant.second.id)
+    get "/api/v1/invoices/find_all?merchant_id=#{Merchant.first.id}"
 
     expect(response).to be_success
 
     invoices = JSON.parse(response.body)
     expect(invoices.count).to eq(3)
+    expect(invoices.sample["merchant_id"]).to eq(Merchant.first.id)
+  end
+  it "finds multiple invoices by status" do
+    create_list(:invoice, 3, status: "shipped")
+    create(:invoice, status: "cancelled")
+    get "/api/v1/invoices/find_all?status=shipped"
+
+    expect(response).to be_success
+
+    invoices = JSON.parse(response.body)
+    expect(invoices.count).to eq(3)
+    expect(invoices.sample["status"]).to eq("shipped")
+  end
+  it "finds multiple invoices by status" do
+    create_list(:invoice, 3, status: "shipped")
+    create(:invoice, status: "cancelled")
+    get "/api/v1/invoices/find_all?status=shipped"
+
+    expect(response).to be_success
+
+    invoices = JSON.parse(response.body)
+    expect(invoices.count).to eq(3)
+    expect(invoices.sample["status"]).to eq("shipped")
+  end
+  it "finds multiple invoices by created_at" do
+    create_list(:invoice, 3, created_at: "15 May 2017")
+    create(:invoice, created_at: "16 May 2017")
+    get "/api/v1/invoices/find_all?created_at=15 May 2017"
+
+    expect(response).to be_success
+
+    invoices = JSON.parse(response.body)
+    expect(invoices.count).to eq(3)
+  end
+  it "finds multiple invoices by updated_at" do
+    create_list(:invoice, 3, updated_at: "15 May 2017")
+    create(:invoice, updated_at: "16 May 2017")
+    get "/api/v1/invoices/find_all?updated_at=15 May 2017"
+
+    expect(response).to be_success
+
+    invoices = JSON.parse(response.body)
+    expect(invoices.count).to eq(3)
+  end
+  it "finds a random invoice" do
+    create_list(:invoice, 3)
+
+    get "/api/v1/invoices/random.json"
```

Comment:
the .json at the end of the url is only necessary if we decide to go with Jbuilder